pool/database/seed/seed_user.ml

```diff
@@ -63,23 +63,51 @@ let create_rand_persons n_persons =
   else failwith "Could not find or decode any person data."
 ;;
 
-let create_persons n_persons =
+let create_persons db_label n_persons =
+  let open CCFun in
+  let open CCList in
+  let open Utils.Lwt_result.Infix in
   let chunk_size = 100 in
-  let sum = CCList.fold_left ( + ) 0 in
-  let%lwt persons_chunked =
-    n_persons
-    |> CCFun.flip CCList.repeat [ 1 ]
-    |> CCList.chunks chunk_size
-    |> CCList.map sum
-    |> Lwt_list.map_s create_rand_persons
+  let sum = fold_left ( + ) 0 in
+  let%lwt contacts =
+    Contact.find_all db_label ()
+    ||> map (Contact.email_address %> User.EmailAddress.value)
+  in
+  let%lwt admins = Admin.find_all db_label () ||> map Admin.email in
+  let flatten_filter_combine a b =
+    let filter_existing =
+      filter (fun { email; _ } -> mem email (admins @ contacts) |> not)
+    in
+    (flatten b |> filter_existing) @ a
+    |> uniq ~eq:(fun p1 p2 -> CCString.equal p1.email p2.email)
+  in
+  let generate_persons =
+    flip repeat [ 1 ]
+    %> chunks chunk_size
+    %> map sum
+    %> Lwt_list.map_s create_rand_persons
+  in
+  let rec persons_chunked acc n_persons =
+    let current_count = length acc in
+    if current_count < n_persons
+    then (
+      Logs.info (fun m ->
+        m "Seed: generating person data (%i/%i)" current_count n_persons);
+      let%lwt iter =
+        n_persons - current_count
+        |> generate_persons
+        ||> flatten_filter_combine acc
+      in
+      persons_chunked iter n_persons)
+    else take_drop n_persons acc |> fst |> Lwt.return
   in
-  CCList.flatten persons_chunked |> Lwt.return
+  persons_chunked [] n_persons
```

Comment:
I think this function will always generate `n_persons` in one chunk, as `n_persons - current_count` in the first iteration is always `n_persons`? Or am I wrong?

```
  let rec persons_chunked acc chunk_size =
    let current_count = length acc in
    if current_count < n_persons
    then (
      Logs.info (fun m ->
        m "Seed: generating person data (%i/%i)" current_count n_persons);
      let%lwt iter =
        CCInt.min (n_persons - current_count) chunk_size
        |> generate_persons
        ||> flatten_filter_combine acc
      in
      persons_chunked iter chunk_size)
    else take_drop n_persons acc |> fst |> Lwt.return
  in
  persons_chunked [] chunk_size
```

Reply:
Yes, in the first run, it's usually always `n_persons`.
But `chunk_size` is needed for the API calls, which is handled by the `generate_persons` function.

Reply:
I removed the unneeded variable forwarding of `n_persons` and directly use the main function input.

Reply:
I tested my change before and it worked :) I passed `CCInt.min (n_persons - current_count) chunk_size` to generate_persons function.

But now the functionality is clearer.